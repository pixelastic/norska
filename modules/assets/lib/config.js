module.exports = {
  files: [
    // Static files
    '**/*.{html,txt}',
    // Images
    '**/*.{ico,jpg,jpeg,gif,png,svg}',
    // Fonts
    '**/*.{eot,otf,ttf,woff,woff2}',
    // Documents
    '**/*.pdf',
    // Netlify
    '_redirects',
    '_headers',
    'netlify.toml',
  ],
};
