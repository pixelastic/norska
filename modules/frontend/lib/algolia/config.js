module.exports = {
  options: {
    routerIgnore: [], // List of parameters to ignore in the router (will not be added to the URL)
  },
};
