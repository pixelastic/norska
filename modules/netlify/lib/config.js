module.exports = {
  deploy: {
    // Deploy again if any of those files were changed since the last deploy:
    // - All path are relative to the repo root
    // - <projectRoot> is where the package.json holding norska is
    // - <from> is the source folder, relative to the <projectRoot>
    files: [
      'lambda/**/*',
      'netlify.toml',
      '.nvmrc',
      '<projectRoot>/norska.config.js',
      '<projectRoot>/tailwind.config.js',
      '<from>/**/*',
    ],
    // If any of those keys in package.json  has been modified since the last
    // deploy, we deploy again
    keys: ['dependencies', 'scripts.build:prod'],
  },
};
