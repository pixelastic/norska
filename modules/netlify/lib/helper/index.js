const NetlifyAPI = require('netlify');
const config = require('norska-config');
const exists = require('firost/exists');
const readJson = require('firost/readJson');
const got = require('golgoth/got');
const _ = require('golgoth/lodash');
module.exports = {
  /**
   * Returns the Netlify token saved in ENV
   * @returns {string} The Netlify token
   **/
  token() {
    return this.getEnvVar('NETLIFY_AUTH_TOKEN');
  },
  /**
   * Check if a Netlify token is available
   * @returns {boolean} True if a token is defined
   **/
  hasToken() {
    return !!this.token();
  },
  /**
   * Returns a Netlify API Client
   * Always returns the same instance
   * @returns {object} NetlifyAPI instance
   **/
  apiClient() {
    if (!this.__apiClient) {
      const token = this.token();
      this.__apiClient = new NetlifyAPI(token);
    }
    return this.__apiClient;
  },
  /**
   * Check if the current code is running on Netlify servers
   * @returns {boolean} True if running remotely, false otherwise
   **/
  isRunningRemotely() {
    return !!this.getEnvVar('NETLIFY');
  },
  /**
   * Returns the current siteId
   * @returns {string} Netlify Site ID
   **/
  async siteId() {
    if (this.__siteId) {
      return this.__siteId;
    }

    // Remotely: saved as SITE_ID
    if (this.isRunningRemotely()) {
      return this.getEnvVar('SITE_ID');
    }

    // Locally: Saved in .netlify/state.json
    return await this.localSiteId();
  },
  /**
   * Check if the current repo is linked to a remote site
   * @returns {boolean} True if linked
   **/
  async isLinkedLocally() {
    return !!(await this.localSiteId());
  },
  /**
   * Returns the siteId from the local filesystem
   * @returns {string} Netlify siteId
   **/
  async localSiteId() {
    const netlifyStateFile = config.rootPath('.netlify/state.json');
    if (!(await exists(netlifyStateFile))) {
      return false;
    }
    const netlifyState = await readJson(netlifyStateFile);
    return netlifyState.siteId;
  },
  /**
   * Returns the current Netlify site name
   * @returns {string} Site name
   **/
  async siteName() {
    const siteId = await this.siteId();
    const client = this.apiClient();
    const response = await client.getSite({ site_id: siteId });
    return response.name;
  },
  /**
   * Returns an ENV var by name.
   * @param {string} key Name of the var
   * @returns {string} Value of the var
   **/
  getEnvVar(key) {
    return process.env[key];
  },
  /**
   * Make a raw API call to the Netlify API without using their API client
   * because some endpoints are not implemented
   * @param {string} apiPath Path of the api, after the api/v1 part
   * @param {object} userOptions Options to pass to the internal got call
   * @returns {object} Parsed JSON response or raw response
   **/
  async rawApiCall(apiPath, userOptions = {}) {
    const token = this.token();
    const apiUrl = `https://api.netlify.com/api/v1/${apiPath}`;
    const defaultOptions = {
      method: 'get',
      headers: {
        authorization: `Bearer ${token}`,
      },
    };
    const options = _.merge({}, defaultOptions, userOptions);
    const response = await got(apiUrl, options);
    try {
      return JSON.parse(response.body);
    } catch (err) {
      return response.body;
    }
  },
};
