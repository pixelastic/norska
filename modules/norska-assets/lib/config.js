module.exports = {
  files: [
    // Static files
    '**/*.{html,txt}',
    // Images
    '**/*.{ico,jpg,gif,png,svg}',
    // Fonts
    '**/*.{eot,otf,ttf,woff,woff2}',
    // Documents
    '**/*.pdf',
    // Netlify
    '_redirects',
    '_headers',
    'netlify.toml',
  ],
};
