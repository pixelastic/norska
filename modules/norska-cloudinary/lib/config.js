module.exports = {
  enable: true, // Set to false to disable the module
  bucketName: null, // bucketName used in urls
};
