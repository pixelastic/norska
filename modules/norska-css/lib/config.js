module.exports = {
  input: 'style.css',
};
