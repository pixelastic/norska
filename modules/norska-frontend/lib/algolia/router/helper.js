const credentials = require('../credentials.js');
module.exports = {
  metaKeys: ['page', 'query'],
  indexKeys: ['sortBy', 'index'],
  filterKeys: ['refinementList', 'range'],
  rangePattern: /^{(.*)}/,
  refinementListPattern: /^\[(.*)\]/,
  /**
   * Converts a key and value pair to a string suitable to be put in the URL
   * @param {*} value Value of the filters
   * @param {string} key Key name
   * @returns {string} URL part
   **/
  keyValueToString(value, key) {
    const isMeta = this.metaKeys.includes(key);
    const isIndex = this.indexKeys.includes(key);

    if (isMeta || isIndex) {
      return `${key}:${encodeURIComponent(value)}`;
    }

    const isFilter = Array.isArray(value);

    // Filters
    if (isFilter) {
      const encodedValue = value.map(encodeURIComponent);
      const joinedValue = encodedValue.sort().join(',');
      return `${key}:[${joinedValue}]`;
    }

    // Range
    const isRange = value.includes(':');
    if (isRange) {
      const urlValue = value.replace(':', ',');
      return `${key}:{${urlValue}}`;
    }

    return 'TODO';
  },
  /**
   * Converts a key:value hash string into a filter object
   * @param {string} input key:value hash string
   * @returns {object} Filter object
   **/
  stringToFilterObject(input) {
    const [rawKey, rawValue] = input.split(':');
    let key = rawKey;
    let value = rawValue;

    // Stop if a filter than changes the index
    const isIndex = this.indexKeys.includes(key);
    if (isIndex) {
      return {};
    }

    // Pages must be converted to number
    const isPage = key === 'page';
    if (isPage) {
      value = parseInt(value);
    }

    // Meta keys are at the root
    const isMeta = this.metaKeys.includes(key);
    if (isMeta) {
      return {
        [key]: decodeURIComponent(value),
      };
    }

    // Refinement list
    const isFilter = this.refinementListPattern.test(value);
    if (isFilter) {
      const capture = this.refinementListPattern.exec(value);
      value = capture[1]
        .split(',')
        .map(decodeURIComponent)
        .sort();
      return {
        refinementList: {
          [key]: value,
        },
      };
    }

    // Range
    const isRange = this.rangePattern.test(value);
    if (isRange) {
      const capture = this.rangePattern.exec(value);
      value = capture[1].replace(',', ':');
      return {
        range: {
          [key]: value,
        },
      };
    }
  },
  /**
   * Return the name of the index base on the location hash
   * Default to base index, unless index: is set, or sortBy: is set
   * @param {string} locationHash location hash to decode
   * @returns {string} name of the index
   */
  indexNameFromLocationHash(locationHash) {
    const stringFilters = locationHash.replace(/^#/, '').split('/');
    const filters = {};
    stringFilters.forEach(filter => {
      const [key, value] = filter.split(':');
      filters[key] = value;
    });

    const baseIndex = this.indexName();
    const sortByKey = filters.sortBy;
    const indexKey = filters.index;
    if (sortByKey) {
      return `${baseIndex}_${sortByKey}`;
    }
    if (indexKey) {
      return indexKey;
    }
    return baseIndex;
  },
  /**
   * Return the current url
   * @returns {string} Current url, without query string nor hash
   **/
  currentUrl() {
    const fullUrl = `${location.origin}${location.pathname}`;
    return fullUrl.replace(/index\.html$/, '');
  },
  /**
   * Returns the index name
   * @returns {string} Name of the main index
   **/
  indexName() {
    return credentials.indexName();
  },
  /**
   * Checks if a given index name seems to be a replica
   * Replicas start with the base index name as prefix
   * @param {string} indexName name of the index to test
   * @returns {boolean} true if seems like a replica
   **/
  isReplica(indexName) {
    const baseIndexName = this.indexName();
    return indexName.startsWith(baseIndexName);
  },
  /**
   * Return the short replica name (minus the base index prefix)
   * @param {string} indexName name of the replica index
   * @returns {string} short name of the replica
   **/
  replicaShortname(indexName) {
    const baseIndexName = this.indexName();
    const regexp = new RegExp(`^${baseIndexName}_`);
    return indexName.replace(regexp, '');
  },
};
