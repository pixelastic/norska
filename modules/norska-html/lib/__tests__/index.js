import module from '../index';
import config from 'norska-config';
import helper from 'norska-helper';
import firost from 'firost';

describe('norska-html', () => {
  beforeEach(async () => {
    await config.init({
      from: './fixtures/src',
      to: './tmp/norska-html',
    });
  });
  describe('pugFiles', () => {
    it('should find pug file in source', async () => {
      const actual = await module.pugFiles();

      expect(actual).toContain(config.fromPath('foo.pug'));
    });
    it('should find pug file in sub directory of source', async () => {
      const actual = await module.pugFiles();

      expect(actual).toContain(config.fromPath('subdir/foo.pug'));
    });
    it('should not find pug file in root', async () => {
      const actual = await module.pugFiles();

      expect(actual).not.toContain(config.rootPath('root.pug'));
    });
    it('should not find layout pug file', async () => {
      const actual = await module.pugFiles();

      expect(actual).not.toContain(config.fromPath('_layouts/default.pug'));
    });
    it('should not find include pug file', async () => {
      const actual = await module.pugFiles();

      expect(actual).not.toContain(config.fromPath('_includes/mixins.pug'));
    });
  });
  describe('compile', () => {
    it('should fail if file is not in the source folder', async () => {
      jest.spyOn(helper, 'consoleWarn').mockReturnValue();
      const input = '/nope/foo.pug';

      const actual = await module.compile(input);

      expect(actual).toEqual(false);
      expect(helper.consoleWarn).toHaveBeenCalled();
    });
    it('should compile foo.pug', async () => {
      const input = 'foo.pug';
      const output = 'foo.html';

      await module.compile(input);
      const actual = await firost.read(config.toPath(output));

      expect(actual).toMatchSnapshot();
    });
    it('should compile ./subdir/foo.pug', async () => {
      const input = 'subdir/foo.pug';
      const output = 'subdir/foo.html';

      await module.compile(input);
      const actual = await firost.read(config.toPath(output));

      expect(actual).toMatchSnapshot();
    });
    it('should compile ./subdir/deep/foo.pug', async () => {
      const input = 'subdir/deep/foo.pug';
      const output = 'subdir/deep/foo.html';

      await module.compile(input);
      const actual = await firost.read(config.toPath(output));

      expect(actual).toMatchSnapshot();
    });
    it('should compile ./with-data.pug', async () => {
      const input = 'with-data.pug';
      const output = 'with-data.html';

      await module.compile(input);
      const actual = await firost.read(config.toPath(output));

      expect(actual).toMatchSnapshot();
    });
    describe('layouts', () => {
      it('absolute in root', async () => {
        const input = 'with-layout-absolute.pug';
        const output = 'with-layout-absolute.html';

        await module.compile(input);
        const actual = await firost.read(config.toPath(output));

        expect(actual).toMatchSnapshot();
      });
      it('relative in root', async () => {
        const input = 'with-layout-relative.pug';
        const output = 'with-layout-relative.html';

        await module.compile(input);
        const actual = await firost.read(config.toPath(output));

        expect(actual).toMatchSnapshot();
      });
      it('absolute in subdir', async () => {
        const input = 'subdir/with-layout-absolute.pug';
        const output = 'subdir/with-layout-absolute.html';

        await module.compile(input);
        const actual = await firost.read(config.toPath(output));

        expect(actual).toMatchSnapshot();
      });
      it('relative in subdir', async () => {
        const input = 'subdir/with-layout-relative.pug';
        const output = 'subdir/with-layout-relative.html';

        await module.compile(input);
        const actual = await firost.read(config.toPath(output));

        expect(actual).toMatchSnapshot();
      });
    });
    describe('paths', () => {
      it('should have paths for ./with-paths.pug', async () => {
        const input = 'with-paths.pug';
        const output = 'with-paths.html';

        await module.compile(input);
        const actual = await firost.read(config.toPath(output));

        expect(actual).toMatchSnapshot();
      });
      it('should have paths for ./subdir/with-paths.pug', async () => {
        const input = 'subdir/with-paths.pug';
        const output = 'subdir/with-paths.html';

        await module.compile(input);
        const actual = await firost.read(config.toPath(output));

        expect(actual).toMatchSnapshot();
      });
      it('should have paths for ./subdir/deep/with-paths.pug', async () => {
        const input = 'subdir/deep/with-paths.pug';
        const output = 'subdir/deep/with-paths.html';

        await module.compile(input);
        const actual = await firost.read(config.toPath(output));

        expect(actual).toMatchSnapshot();
      });
    });
  });
  describe('getPaths', () => {
    describe('basename', () => {
      it('foo.html => foo.html', () => {
        const actual = module.getPaths(config.toPath('foo.html'));

        expect(actual).toHaveProperty('basename', 'foo.html');
      });
      it('subdir/foo.html => foo.html', () => {
        const actual = module.getPaths(config.toPath('subdir/foo.html'));

        expect(actual).toHaveProperty('basename', 'foo.html');
      });
      it('subdir/deep/foo.html => foo.html', () => {
        const actual = module.getPaths(config.toPath('subdir/deep/foo.html'));

        expect(actual).toHaveProperty('basename', 'foo.html');
      });
    });
    describe('dirname', () => {
      it('foo.html => ', () => {
        const actual = module.getPaths(config.toPath('foo.html'));

        expect(actual).toHaveProperty('dirname', '');
      });
      it('subdir/foo.html => subdir', () => {
        const actual = module.getPaths(config.toPath('subdir/foo.html'));

        expect(actual).toHaveProperty('dirname', 'subdir');
      });
      it('subdir/deep/foo.html => subdir', () => {
        const actual = module.getPaths(config.toPath('subdir/deep/foo.html'));

        expect(actual).toHaveProperty('dirname', 'subdir/deep');
      });
    });
    describe('toRoot', () => {
      it('foo.html => .', () => {
        const actual = module.getPaths(config.toPath('foo.html'));

        expect(actual).toHaveProperty('toRoot', '.');
      });
      it('subdir/foo.html => ..', () => {
        const actual = module.getPaths(config.toPath('subdir/foo.html'));

        expect(actual).toHaveProperty('toRoot', '..');
      });
      it('subdir/deep/foo.html => ..', () => {
        const actual = module.getPaths(config.toPath('subdir/deep/foo.html'));

        expect(actual).toHaveProperty('toRoot', '../..');
      });
    });
  });
  describe('run', () => {
    beforeAll(async () => {
      await config.init({
        from: './fixtures/src',
        to: './tmp/norska-html',
      });
      await firost.emptyDir('./tmp/norska-html');
      await module.run();
    });
    describe('excluded files', () => {
      it('_includes/mixins.html should not exist', async () => {
        const input = '_includes/mixins.html';
        const actual = await firost.exist(config.toPath(input));

        expect(actual).toEqual(false);
      });
      it('_layouts/default.html should not exist', async () => {
        const input = '_layouts/default.html';
        const actual = await firost.exist(config.toPath(input));

        expect(actual).toEqual(false);
      });
    });
    describe('simple files', () => {
      it('foo.html', async () => {
        const input = 'foo.html';
        const actual = await firost.read(config.toPath(input));

        expect(actual).toMatchSnapshot();
      });
      it('subdir/foo.html', async () => {
        const input = 'subdir/foo.html';
        const actual = await firost.read(config.toPath(input));

        expect(actual).toMatchSnapshot();
      });
      it('subdir/deep/foo.html', async () => {
        const input = 'subdir/deep/foo.html';
        const actual = await firost.read(config.toPath(input));

        expect(actual).toMatchSnapshot();
      });
    });
    describe('with paths', () => {
      it('with-paths.html', async () => {
        const input = 'with-paths.html';
        const actual = await firost.read(config.toPath(input));

        expect(actual).toMatchSnapshot();
      });
      it('subdir/with-paths.html', async () => {
        const input = 'subdir/with-paths.html';
        const actual = await firost.read(config.toPath(input));

        expect(actual).toMatchSnapshot();
      });
      it('subdir/deep/with-paths.html', async () => {
        const input = 'subdir/deep/with-paths.html';
        const actual = await firost.read(config.toPath(input));

        expect(actual).toMatchSnapshot();
      });
    });
    describe('with layout', () => {
      it('with-layout-absolute.html', async () => {
        const input = 'with-layout-absolute.html';
        const actual = await firost.read(config.toPath(input));

        expect(actual).toMatchSnapshot();
      });
      it('subdir/with-layout-absolute.html', async () => {
        const input = 'subdir/with-layout-absolute.html';
        const actual = await firost.read(config.toPath(input));

        expect(actual).toMatchSnapshot();
      });
      it('with-layout-relative.html', async () => {
        const input = 'with-layout-relative.html';
        const actual = await firost.read(config.toPath(input));

        expect(actual).toMatchSnapshot();
      });
      it('subdir/with-layout-relative.html', async () => {
        const input = 'subdir/with-layout-relative.html';
        const actual = await firost.read(config.toPath(input));

        expect(actual).toMatchSnapshot();
      });
    });
  });
});
