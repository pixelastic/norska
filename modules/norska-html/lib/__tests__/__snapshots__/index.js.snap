// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`norska-html compile layouts absolute in root 1`] = `"<html><head><title>Title</title></head><body><p>In a layout</p></body></html>"`;

exports[`norska-html compile layouts absolute in subdir 1`] = `"<html><head><title>Title</title></head><body><p>In a layout</p></body></html>"`;

exports[`norska-html compile layouts relative in root 1`] = `"<html><head><title>Title</title></head><body><p>In a layout</p></body></html>"`;

exports[`norska-html compile layouts relative in subdir 1`] = `"<html><head><title>Title</title></head><body><p>In a layout</p></body></html>"`;

exports[`norska-html compile paths should have paths for ./subdir/deep/with-paths.pug 1`] = `"<p>with-paths.html</p><p>subdir/deep</p><p>../..</p>"`;

exports[`norska-html compile paths should have paths for ./subdir/with-paths.pug 1`] = `"<p>with-paths.html</p><p>subdir</p><p>..</p>"`;

exports[`norska-html compile paths should have paths for ./with-paths.pug 1`] = `"<p>with-paths.html</p><p></p><p>.</p>"`;

exports[`norska-html compile should compile ./subdir/deep/foo.pug 1`] = `"<p>Subdir Deep Foo</p>"`;

exports[`norska-html compile should compile ./subdir/foo.pug 1`] = `"<p>Subdir Foo</p>"`;

exports[`norska-html compile should compile ./with-data.pug 1`] = `"<p>bar</p>"`;

exports[`norska-html compile should compile foo.pug 1`] = `"<p>Foo</p>"`;

exports[`norska-html run simple files foo.html 1`] = `"<p>Foo</p>"`;

exports[`norska-html run simple files subdir/deep/foo.html 1`] = `"<p>Subdir Deep Foo</p>"`;

exports[`norska-html run simple files subdir/foo.html 1`] = `"<p>Subdir Foo</p>"`;

exports[`norska-html run with layout subdir/with-layout-absolute.html 1`] = `"<html><head><title>Title</title></head><body><p>In a layout</p></body></html>"`;

exports[`norska-html run with layout subdir/with-layout-relative.html 1`] = `"<html><head><title>Title</title></head><body><p>In a layout</p></body></html>"`;

exports[`norska-html run with layout with-layout-absolute.html 1`] = `"<html><head><title>Title</title></head><body><p>In a layout</p></body></html>"`;

exports[`norska-html run with layout with-layout-relative.html 1`] = `"<html><head><title>Title</title></head><body><p>In a layout</p></body></html>"`;

exports[`norska-html run with paths subdir/deep/with-paths.html 1`] = `"<p>with-paths.html</p><p>subdir/deep</p><p>../..</p>"`;

exports[`norska-html run with paths subdir/with-paths.html 1`] = `"<p>with-paths.html</p><p>subdir</p><p>..</p>"`;

exports[`norska-html run with paths with-paths.html 1`] = `"<p>with-paths.html</p><p></p><p>.</p>"`;
